torch/nn/quantizable/modules/activation.py

```diff
@@ -0,0 +1,368 @@
+    representation subspaces.
+    See reference: Attention Is All You Need
+
+    .. math::
+        \text{MultiHead}(Q, K, V) = \text{Concat}(head_1,\dots,head_h)W^O
+        \text{where} head_i = \text{Attention}(QW_i^Q, KW_i^K, VW_i^V)
+
+    Args:
+        embed_dim: total dimension of the model.
+        num_heads: parallel attention heads.
+        dropout: a Dropout layer on attn_output_weights. Default: 0.0.
+        bias: add bias as module parameter. Default: True.
+        add_bias_kv: add bias to the key and value sequences at dim=0.
+        add_zero_attn: add a new batch of zeros to the key and
+                       value sequences at dim=1.
+        kdim: total number of features in key. Default: None.
+        vdim: total number of features in value. Default: None.
+
+        Note: if kdim and vdim are None, they will be set to embed_dim such that
+        query, key, and value have the same number of features.
+
+    Examples::
+
+        >>> import torch.nn.quantizable as nnqa
+        >>> multihead_attn = nnqa.MultiheadAttention(embed_dim, num_heads)
+        >>> attn_output, attn_output_weights = multihead_attn(query, key, value)
+
+    Note::
+        Please, follow the quantization flow to convert the quantizable MHA.
+    """
+    def __init__(self, embed_dim: int, num_heads: int,
+                 dropout: float = 0., bias: bool = True,
+                 add_bias_kv: bool = False, add_zero_attn: bool = False,
+                 kdim: int = None, vdim: int = None):
+        super(MultiheadAttention, self).__init__(embed_dim, num_heads, dropout,
+                                                 bias, add_bias_kv,
+                                                 add_zero_attn, kdim, vdim)
+        self.linear_Q = nn.Linear(self.embed_dim, self.embed_dim, bias=bias)
+        self.linear_K = nn.Linear(self.kdim, self.embed_dim, bias=bias)
+        self.linear_V = nn.Linear(self.vdim, self.embed_dim, bias=bias)
+
+        # TODO: The use of the `_LinearWithBias` increases the quantization noise
+        # The `out_proj` in the parent is ``_LinearWithBias`, so need to ignore
+        # the type for mypy not to complain.
+        self.out_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=True)  # type: ignore
+
+        # Functionals
+        self.q_scaling_product = nnq.FloatFunctional()
+
+        # Quant/Dequant
+        self.quant_attn_output = torch.quantization.QuantStub()
+        self.quant_attn_output_weights = torch.quantization.QuantStub()
+        self.dequant_q = torch.quantization.DeQuantStub()
+        self.dequant_k = torch.quantization.DeQuantStub()
+        self.dequant_v = torch.quantization.DeQuantStub()
+
+    def _get_name(self):
+        return 'QuantizableMultiheadAttention'
+
+    @classmethod
+    def from_float(cls, other):
+        assert type(other) == cls._FLOAT_MODULE
+        assert hasattr(other, 'qconfig'), "The float module must have 'qconfig'"
+        # Setting the dropout to 0.0!
```

Comment:
Can we support QAT with this module?  We should not set the dropout to zero here, this will be handled when the model is set to eval.

Reply:
The `prepare_qat` is not implemented, so there is not need for this PR